src/js/hooks/inboundV2/addItems/useInboundAddItemsForm.jsx

```diff
@@ -0,0 +1,566 @@
+
+    if (startIndex !== null && getValues('values.lineItems').length !== getValues().totalCount) {
+      loadMoreRows({ startIndex: startIndex + getValues().pageSize });
+    }
+  };
+
+  const fetchLineItems = async () => {
+    if (queryParams.id) {
+      const url = `${STOCK_MOVEMENT_ITEMS(queryParams.id)}?stepNumber=2`;
+      const response = await apiClient.get(url);
+      setValue('totalCount', response.data.data.length);
+      setLineItems(response, null);
+    }
+  };
+
+  const removeAll = async () => {
+    try {
+      spinner.show();
+      await apiClient.delete(STOCK_MOVEMENT_REMOVE_ALL_ITEMS(queryParams.id));
+      setValue('totalCount', 1);
+      setValue('currentLineItems', []);
+      setValue('values', { ...getValues('values'), lineItems: new Array(1).fill({ sortOrder: 100 }) });
+      await fetchLineItems();
+    } finally {
+      spinner.hide();
+    }
+  };
+
+  const previousPage = async () => {
+    trigger();
+    if (isValid) {
+      try {
+        spinner.show();
+        await saveRequisitionItemsInCurrentStep(getValues().values.lineItems);
+        previous();
+      } finally {
+        spinner.hide();
+      }
+    } else {
+      confirmAlert({
+        title: translate('react.stockMovement.confirmPreviousPage.label', 'Validation error'),
+        message: translate('react.stockMovement.confirmPreviousPage.message.label', 'Cannot save due to validation error on page'),
+        buttons: [
+          {
+            label: translate('react.stockMovement.confirmPreviousPage.correctError.label', 'Correct error'),
+          },
+          {
+            label: translate('react.stockMovement.confirmPreviousPage.continue.label', 'Continue (lose unsaved work)'),
+            onClick: () => previous(),
+          },
+        ],
+      });
+    }
+  };
+
+  const fetchAddItemsPageData = async () => {
+    if (queryParams.id) {
+      const response = await apiClient.get(STOCK_MOVEMENT_BY_ID(queryParams.id));
+      const { totalCount, data } = response.data;
+      const transformedData = {
+        ...data,
+        lineItems: data.lineItems?.map(transformLineItem),
+      };
+
+      setValue('values', transformedData);
+      setValue('totalCount', totalCount || 1);
+    }
+  };
+
+  const fetchData = async () => {
+    setLoading(true);
+    spinner.show();
+    try {
+      await fetchAddItemsPageData();
+      if (getValues().isPaginated) {
+        await fetchLineItems();
+      }
+    } finally {
+      setLoading(false);
+      spinner.hide();
+    }
+  };
+
+  useEffect(() => {
+    if (queryParams.step === InboundV2Step.ADD_ITEMS) {
+      fetchData();
+    }
+  }, [queryParams.step]);
+
+  return {
+    control,
+    getValues,
+    setValue,
+    handleSubmit,
+    errors,
+    isValid,
+    trigger,
+    loading,
+    nextPage,
+    save,
+    removeItem,
+    updateTotalCount,
+    removeAll,
+    saveAndExit,
+    previousPage,
+  };
+};
+
+export default useInboundAddItemsForm;
```

Comment:
In my opinion we should make those methods more readable if we refactor the whole workflow. Most of them are just copied and remain difficult to read and understand. @alannadolny and I can assist you with that, if you have doubts and questions.